### ISSUE TITLE
DoesForEach() extension method

### ISSUE THREAD

**Issue by david-driscoll** [cake-build/cake#1761]
I ran into a scenario where I wanted to run a set of actions over a collection, even in the event that one of more fails.  The main one is running unit tests, but there are other scenarios like handling many files with a common action, that I've found it handy.

For my code it's replaced anywhere I would use a `foreach` at the top of my `Does()` statement, hence calling the extension method `DoesForEach()`.

```csharp
public static CakeTaskBuilder<ActionTask> DoesForEach<T>(this CakeTaskBuilder<ActionTask> builder, IEnumerable<T> items, Action<T> action)
{
    return DoesForEach(builder, () => items, (i, c) => action(i));
}

public static CakeTaskBuilder<ActionTask> DoesForEach<T>(this CakeTaskBuilder<ActionTask> builder, IEnumerable<T> items, Action<T, ICakeContext> action)
{
    return DoesForEach(builder, () => items, action);
}
public static CakeTaskBuilder<ActionTask> DoesForEach<T>(this CakeTaskBuilder<ActionTask> builder, Func<IEnumerable<T>> items, Action<T> action)
{
    return DoesForEach(builder, items, (i, c) => action(i));
}

public static CakeTaskBuilder<ActionTask> DoesForEach<T>(this CakeTaskBuilder<ActionTask> builder, Func<IEnumerable<T>> items, Action<T, ICakeContext> action)
{
    builder.Does(c =>
    {
        var exceptions = new List<Exception>();
        foreach (var item in items())
        {
            try
            {
                action(item, c);
            }
            catch (Exception e)
            {
                exceptions.Add(e);
            }
        }
        if (exceptions.Any()) throw new AggregateException("Task failed with following exceptions", exceptions);
    });
    return builder;
}
```

**Comment by devlead** [comment]
Sounds like a reasonable addition to cake, would you be willing to send in a PR for review?

**Comment by david-driscoll** [comment]
Closing merged in #1767
